packages/block-library/src/cover/edit.js

```diff
@@ -171,281 +139,249 @@ function ResizableCover( {
-			onResize={ onResizeEvent }
-			onResizeStop={ onResizeStopEvent }
+			onResizeStart={ ( event, direction, elt ) => {
+				onResizeStart( elt.clientHeight );
+				onResize( elt.clientHeight );
+			} }
+			onResize={ ( event, direction, elt ) => {
+				onResize( elt.clientHeight );
+				if ( ! isResizing ) {
+					setIsResizing( true );
+				}
+			} }
+			onResizeStop={ ( event, direction, elt ) => {
+				onResizeStop( elt.clientHeight );
+				setIsResizing( false );
+			} }
 			minHeight={ COVER_MIN_HEIGHT }
 		>
 			{ children }
 		</ResizableBox>
 	);
 }
 
-class CoverEdit extends Component {
-	constructor() {
-		super( ...arguments );
-		this.state = {
-			isDark: false,
-			temporaryMinHeight: null,
-		};
-		this.imageRef = createRef();
-		this.videoRef = createRef();
-		this.changeIsDarkIfRequired = this.changeIsDarkIfRequired.bind( this );
-		this.onUploadError = this.onUploadError.bind( this );
-	}
-
-	componentDidMount() {
-		this.handleBackgroundMode();
-	}
-
-	componentDidUpdate( prevProps ) {
-		this.handleBackgroundMode( prevProps );
-	}
+function onCoverSelectMedia( setAttributes ) {
+	return ( media ) => {
+		if ( ! media || ! media.url ) {
+			setAttributes( { url: undefined, id: undefined } );
+			return;
+		}
+		let mediaType;
+		// for media selections originated from a file upload.
+		if ( media.media_type ) {
+			if ( media.media_type === IMAGE_BACKGROUND_TYPE ) {
+				mediaType = IMAGE_BACKGROUND_TYPE;
+			} else {
+				// only images and videos are accepted so if the media_type is not an image we can assume it is a video.
+				// Videos contain the media type of 'file' in the object returned from the rest api.
+				mediaType = VIDEO_BACKGROUND_TYPE;
+			}
+		} else { // for media selections originated from existing files in the media library.
+			if (
+				media.type !== IMAGE_BACKGROUND_TYPE &&
+				media.type !== VIDEO_BACKGROUND_TYPE
+			) {
+				return;
+			}
+			mediaType = media.type;
+		}
 
-	onUploadError( message ) {
-		const { noticeOperations } = this.props;
-		noticeOperations.removeAllNotices();
-		noticeOperations.createErrorNotice( message );
-	}
+		setAttributes( {
+			url: media.url,
+			id: media.id,
+			backgroundType: mediaType,
+			...( mediaType === VIDEO_BACKGROUND_TYPE ?
+				{ focalPoint: undefined, hasParallax: undefined } :
+				{}
+			),
+		} );
+	};
+}
 
-	render() {
-		const {
-			attributes,
-			setAttributes,
-			isSelected,
-			className,
-			noticeUI,
-			overlayColor,
-			setOverlayColor,
-			toggleSelection,
-		} = this.props;
-		const {
-			backgroundType,
-			customGradient,
-			dimRatio,
-			focalPoint,
-			hasParallax,
-			id,
-			minHeight,
-			url,
-		} = attributes;
-		const onSelectMedia = ( media ) => {
-			if ( ! media || ! media.url ) {
-				setAttributes( { url: undefined, id: undefined } );
-				return;
+function useCoverIsDark( url, dimRatio = 50, overlayColor, elementRef ) {
+	const [ isDark, setIsDark ] = useState( false );
+	useEffect(
+		() => {
```

Comment:
Nit: I'd personally write this in the same line as useEffect and the dependencies in the same line as the closing parens. Personal preference of course :)